src/authInterceptor/PublicKeyAuthInterceptor.js

```diff
@@ -0,0 +1,48 @@
+import Connection from '../Connection'
+import AuthInterceptor from './AuthInterceptor'
+
+export default class PublicKeyAuthInterceptor extends AuthInterceptor {
+  constructor(publicAPIKey, options) {
+    super()
+    this.publicAPIKey = publicAPIKey
+    this.options = options
+    this.marginToExpire = 0.05 // a margin to prevent unexpected expiration (5% of the time)
+  }
+
+  interceptRequest(config) {
+    return this.getOneTimeToken()
+      .then(token => {
+        config.headers['x-api-key'] = 'use-scnnr-one-time-token'
+        config.headers['x-scnnr-one-time-token'] = token
+        return config
+      })
+  }
+
+  getOneTimeToken() {
+    return this.retrieveOneTimeToken()
+      .then(() => {
+        const token = this.oneTimeToken
+        delete this.oneTimeToken
+        return token
+      })
+  }
+
+  retrieveOneTimeToken() {
```

Comment:
I wonder if the functions with the naming `verb`+`OneTimeToken()` are a good candidate for making a `OneTimeToken` class (with `OneTimeToken.get()`, etc).
Then the `Interceptor` would only handle logic about intercepting.

Reply:
Good point! I'll try to consider refactoring.